File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
    int maxProfit = 0;
    int n = prices.length;
    int minPrice = Integer.MAX_VALUE;

    for(int i = 0 ; i < n;  i++){
        if(prices[i] < minPrice){
            minPrice = prices[i];
        }

        int profit = prices[i] - minPrice;

        if(profit > maxProfit){
            maxProfit = profit;
        }
    }
    return maxProfit;
        
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let maxProfit = 0;
    let minPrice = Infinity;

    //for loop to iterate through the array to find the min price to buy the stock
    for(let i = 0; i < prices.length;i++){
        if(prices[i] < minPrice){
            minPrice = prices[i];
        }

        let profit = prices[i] - minPrice;

        if(profit > maxProfit){
            maxProfit = profit;
        }
    }
    return maxProfit;
    
};